### DISCUSSION TITLE
How to migrate environments to a new k8s cluster

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#542] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

v12.0.3

### Description

Hi guys,
I need to migrate all non-production environments of my company from a k8s cluster to a new one.
In order to do that I identified the following ways:
* manually from the CMS &rarr; I just need to manually create the namespaces on the new cluster and, after that, I just need to change the `clusterId` in the environments of my project on the CMS, for each project;

* deleting and creating again the environment directly from the Console, for each project &rarr; in this way the console will create the namespaces for me and, since the environments names, IDs and prefixes will be the same, the console should do the rest.

Do you have some other option for me?
Do you have suggestions for me about the options above?
Am I missing something?

Thanks.

**Comment by davidebianchi** [comment, marked as answer]
Hi @silversoul93!

The cluster migration could be an interesting feature to address in the future in the UI. At the moment, the suggested option is to use the Console to delete and create the environment. In this way, Console will manage the creation of the namespace with the correct labels on it (which will identify the company, the project, and other information).

In this way, it should take the release configuration exactly as before.

**Reply by frengooo** [reply]
Thanks @davidebianchi, I'll do it through the Console so.
